test/multivariate/multinomial.jl

```diff
@@ -210,3 +210,8 @@ p_v = [0.1, 0.4, 0.3, 0.8]
 @test_throws DomainError Multinomial(10, p_v)
 @test_throws DomainError Multinomial(10, p_v; check_args=true)
 Multinomial(10, p_v; check_args=false) # should not warn
+
+# check different prob vector types
+p = [0.2, 0.4, 0.3, 0.1]
+@test (rand(Multinomial(10, p)); true)
+@test (rand(Multinomial(10, convert.(Float32, p))); true)
```

Comment:
Can we use better tests that test correctness instead of only "not erroring"? For instance by looping over `Float64` and `Float32` probabilities in the tests above?

Reply:
I'm not sure I follow.
What kind of correctness do you mean?
As for type stability, I'm looking at adding some `@inferred` tests here for that.

Reply:
That the samples follow the desired distribution - basically, the same tests as for Float64.

Reply:
I'd prefer running the existing tests with Float32 as well over a few additional tests.

Reply:
took care of this :heavy_check_mark: